core/interop/IDynamicInterfaceCastable/README.md

```diff
@@ -0,0 +1,67 @@
+`IDynamicInterfaceCastable` Sample
+================
+
+The [`IDynamicInterfaceCastable` API](https://github.com/dotnet/runtime/issues/36654) was introduced in .NET 5 as a way for creating a .NET class that supports interfaces which are not in its metadata.
+
+This sample provides an implementation of `IDynamicInterfaceCastable` that projects a native object as implementing different known managed interfaces. It uses COM conventions (e.g. `QueryInterface`) for interacting with the native object, but does not require the actual COM system.
+
+Note: The sample uses `Marshal` APIs as part of interacting with the native library. The introduction of [C# function pointers](https://github.com/dotnet/csharplang/blob/994c41586e07e38fb6b30902b1715b4025d80c52/proposals/function-pointers.md) will allow that interaction to occur in a more performant manner.
```

Comment:
The function pointers are shipping in previews already. Can the sample be modified to use them?

Reply:
The new syntax for specifying the calling convention (e.g. `cdecl` -> `unmanaged[Cdecl]`) isn't in the shipped previews yet. I was planning on updating the sample once they can use the new syntax.